### PR TITLE
optimize docker image

### PR DESCRIPTION
**Pull request by xprgv**
Decrease docker image size in ~182x times

$ docker images
dicedb/dice-server                  local             f6c14c5a0037   2 minutes ago   5.21MB
dicedb/dice-server                  latest            438e97b919ea   7 weeks ago     948MB

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,12 @@
-# syntax=docker/dockerfile:1
-
-FROM golang:1.17
-
+FROM golang:1.22.5 AS builder
```
Comment (JyotinderSingh) [inline Dockerfile:1]: Let's leave the go version the same as before, we should use the same version for development and deployment.

File: Dockerfile [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 WORKDIR /dicedb
 COPY go.mod go.sum ./
-RUN go mod download
+RUN go mod download -x
 COPY . .
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 \
+    go build -ldflags "-s -w" -o /dicedb/dice-server
 
-RUN CGO_ENABLED=0 GOOS=linux go build -o /dicedb/dice-server
-EXPOSE 7379
```
Comment (JyotinderSingh) [inline Dockerfile:11]: Let's retain the `EXPOSE 7379` line

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for the change, added some minor comments. Could you also attach a screenshot of the docker image successfully starting up and a redis client being able to connect to it?

File: Dockerfile (hunk 1 above)
Comment (JyotinderSingh) [reply Dockerfile:1]: It looks like the repo's version of the dockerfile has been updated, please modify this to line up with the version in go.mod.

**Comment by Maveric-k07** [comment]
@JyotinderSingh can I wrap this up if it is inactive?

**Comment by JyotinderSingh** [comment]
> @JyotinderSingh can I wrap this up if it is inactive?

Go ahead.

**Comment by xprgv** [comment]
> Thanks for the change, added some minor comments. Could you also attach a screenshot of the docker image successfully starting up and a redis client being able to connect to it?

I will try to do it today

**Comment by xprgv** [comment]
Sorry for long inactivity, was far away from my workstation

**Comment by JyotinderSingh** [comment]
> > Thanks for the change, added some minor comments. Could you also attach a screenshot of the docker image successfully starting up and a redis client being able to connect to it?
> 
> I will try to do it today. Sorry for long inactivity, was far away from my workstation

Sure

**Comment by xprgv** [comment]
> Thanks for the change, added some minor comments. Could you also attach a screenshot of the docker image successfully starting up and a redis client being able to connect to it?

image

**Pull request merged**